pandas/tests/arrays/categorical/test_replace.py

```diff
@@ -45,3 +45,41 @@ def test_replace(to_replace, value, expected, flip_categories):
 
     tm.assert_series_equal(expected, result, check_category_order=False)
     tm.assert_series_equal(expected, s, check_category_order=False)
+
+
+@pytest.mark.parametrize(
+    "to_replace,value,input_data,expected_data,inplace",
+    [
+        (r"^\s*$", pd.NA, ["d", "ee", "f", ""], ["d", "ee", "f", pd.NA], False),
+        (r"e{2}", "replace", ["d", "ee", "f", ""], ["d", "replace", "f", ""], False),
+        (r"f", "replace", ["d", "ee", "f", ""], ["d", "ee", "replace", ""], False),
+        (r"^\s*$", pd.NA, ["d", "ee", "f", ""], ["d", "ee", "f", pd.NA], True),
+        (r"e{2}", "replace", ["d", "ee", "f", ""], ["d", "replace", "f", ""], True),
+        (r"f", "replace", ["d", "ee", "f", ""], ["d", "ee", "replace", ""], True),
+    ],
+)
+def test_replace_regex(to_replace, value, input_data, expected_data, inplace):
```

Comment:
if this is testing DataFrame.replace, these belong in tests.frame.methods.test_replace